marshmallow/utils.py

```diff
@@ -334,16 +333,27 @@ def callable_or_raise(obj):
     return obj
 
 
-def get_func_args(func):
-    """Given a callable, return a tuple of argument names. Handles
-    `functools.partial` objects and class-based callables.
-    """
+def get_func_args2(func):
     if isinstance(func, functools.partial):
-        return inspect.getargspec(func.func).args
+        return list(inspect.getargspec(func.func).args)
     if inspect.isfunction(func) or inspect.ismethod(func):
-        return inspect.getargspec(func).args
-    # Callable class
-    return inspect.getargspec(func.__call__).args
+        return list(inspect.getargspec(func).args)
+    if callable(func):
+        return list(inspect.getargspec(func.__call__).args)
+
+
+def get_func_args3(func):
+    if isinstance(func, functools.partial):
+        return list(inspect.signature(func.func).parameters)
+    if inspect.isfunction(func):
+        return list(inspect.signature(func).parameters)
+    if callable(func) or inspect.ismethod(func):
+        return ['self'] + list(inspect.signature(func.__call__).parameters)
+
+
+# Given a callable, return a list of argument names.
+# Handles `functools.partial` objects and callable objects.
+get_func_args = get_func_args2 if PY2 else get_func_args3
```

Comment:
@maximkulkin: yes I think you are right: there are two points to be considered:
1. the branching or PY2 and PY3 should be done in compat.py rather then in utils.py
2. the upward compatibilty: it should be still possible to import get_func_arg from utils.

So I suggest the following solution: 
I put the two functions versions into compat and then I leave the simple relabeling of the function in utils, so that the function can be still imported from utils. I will try to implement this and send another pull request, okay?